tools/build/Library/InstallUnpacker/classes/VersionNumber.php

```diff
@@ -0,0 +1,143 @@
+<?php
+/**
+ * 2007-2018 PrestaShop.
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+class VersionNumber
+{
+    /**
+     * @var float
+     */
+    private $major;
+
+    /**
+     * @var int
+     */
+    private $minor;
+
+    /**
+     * @var int
+     */
+    private $patch;
+
+    /**
+     * @param float $major
+     * @param int $minor
+     * @param int $patch
+     */
+    public function __construct($major, $minor, $patch)
+    {
+        $this->major = $major;
+        $this->minor = $minor;
+        $this->patch = $patch;
+    }
+
+    /**
+     * @param string $versionNumberAsString
+     *
+     * @return VersionNumber
+     */
+    public static function fromString($versionNumberAsString)
+    {
+        $regexp = '#^([\d\.]+)\.(\d+)\.(\d+)$#';
```

Comment:
Incorrect regex:
it allows 
* 1..7.5.0
* 1..............7.5
* 17.18.19..20
* 17.18..19.29

`^(\d+\.\d+|\d+)\.(\d+)\.(\d+)$` This one is maybe better and denied previous example but still allows:
* 1.7.5.0
* 17.20.30.40
* 1.7.5